src/controller/python/BUILD.gn

```diff
@@ -33,6 +33,8 @@ config("controller_wno_deprecate") {
 declare_args() {
   chip_python_version = "0.0"
   chip_python_package_prefix = "chip"
+  chip_python_supports_stack_locking =
+      current_os != "mac" && current_os != "ios"
```

Comment:
This should probably be based on `chip_device_platform == "darwin"`, since that's what determines the PlatformManager implementation used, no?

But even better, shouldn't this depend on `chip_system_config_locking == "none"`?